lib/binaries.sh

```diff
@@ -2,6 +2,27 @@
 
 RESOLVE="$BP_DIR/vendor/resolve-version-$(get_os)"
 
+resolve() {
+  local binary="$1"
+  local versionRequirement="$2"
+  local n=0
+  local output
+
+  # retry this up to 5 times in case of spurious failed API requests
+  until [ $n -ge 5 ]
+  do
+    if output=$($RESOLVE "$binary" "$versionRequirement"); then
+        echo "$output"
+        return 0
+    else
+        n=$((n+1))
+        sleep 1
```

Comment:
might be good to backoff here, even if it's just linear. like `sleep $((n+1))`

Reply:
Good idea 👍